### PR TITLE
[WIP] Azure Monitor: Live Tests for Exporter

### PR DESCRIPTION
**Pull request by TimothyMothra**
**This is still a work in progress. I created this Draft PR to get some early feedback.**

For this proof of concept, I'm using the `AzureMonitorLogExporter` which is still in development. 

This test will:
- provision an Application Insights resource.
- use ILogger to manually log telemetry.
- telemetry will be sent to Azure Monitor ingestion.
- wait for ingestion.
- query Azure Monitor to get the ingestion telemetry.
- validate that actual telemetry matches expected.

### Todo
- Tests need to query Application Insights to inspect telemetry sent to ingestion. Need to work with Pavel to adapt this for mocking.
- Live Tests must wait for Ingestion. Need to review my approach and see if this can be improved.
- Must check in a recorded test file.

### CODE REVIEW

File: sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/tests/Microsoft.OpenTelemetry.Exporter.AzureMonitor.Integration.Tests/FunctionalTests/AzureMonitorLogExporterLiveTests.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+            // ACT
+            var testMessage = "Hello World";
+
+            // TODO: For proper test isolation, we should include a CustomProperty on telemetry items.
+            // CustomProperty should be unique per test method.
+            // This would allow us to run tests in parallel.
+            logger.Log(logLevel: LogLevel.Information, message: testMessage);
+
+            processor.ForceFlush();
```
Comment (reyang) [inline sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/tests/Microsoft.OpenTelemetry.Exporter.AzureMonitor.Integration.Tests/FunctionalTests/AzureMonitorLogExporterLiveTests.cs:53]: Minor, probably put a timeout on `ForceFlush`, and fail the test if `ForceFlush` returned `false`?

File: sdk/monitor/init-test-environment.ps1 [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# This script requires the AZ Powershell module from the PowerShellGallery
+# https://docs.microsoft.com/en-us/powershell/azure/install-az-ps
+# https://www.powershellgallery.com/packages/Az
+
+Connect-AzAccount -Subscription $Subscription
+
+$RootDirectory = (get-item $PSScriptRoot).parent.parent
+$TestResourcesDirectory = Join-Path -Path $RootDirectory -ChildPath eng\common\TestResources
+
+& $TestResourcesDirectory\New-TestResources.ps1 monitor
```
Comment (reyang) [inline sdk/monitor/init-test-environment.ps1:19]: nit, add a blank line before EOF

File: sdk/monitor/test-resources.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        "baseName": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().name]",
+            "metadata": {
+                "description": "The base resource name."
+            }
+        },
+        "tenantId": {
+            "type": "string",
+            "defaultValue": "72f988bf-86f1-41af-91ab-2d7cd011db47",
```
Comment (reyang) [inline sdk/monitor/test-resources.json:14]: curious - who owns this tenant id (and pays for it), do we intend to extract this as a parameter (e.g. GitHub secret)?

File: sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/tests/Microsoft.OpenTelemetry.Exporter.AzureMonitor.Integration.Tests/FunctionalTests/AzureMonitorTestEnvironment.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    public class AzureMonitorTestEnvironment : TestEnvironment
+    {
+        public AzureMonitorTestEnvironment() : base()
+        {
+        }
+
+        public string ConnectionString => GetRecordedVariable(EnvironmentVariableNames.ConnectionString);
+
+        public string InstrumentationKey => GetRecordedVariable(EnvironmentVariableNames.InstrumentationKey);
```
Comment (reyang) [inline sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/tests/Microsoft.OpenTelemetry.Exporter.AzureMonitor.Integration.Tests/FunctionalTests/AzureMonitorTestEnvironment.cs:16]: My gut feeling is that we don't want to use `InstrumentationKey` anymore, and we would stick with ConnectionString. So keeping both could be confusing?

File: sdk/monitor/test-resources.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "The location of the resource. By default, this is the same as the resource group."
+            }
+        }
+    },
+    "variables": {
+        "azureKeyVaultUrl": "[format('https://{0}.vault.azure.net', parameters('baseName'))]"
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/test-resources.json:35]: Variable `azureKeyVaultUrl`, parameters `tenantId` and `testApplicationOid` are not used in resources. It is not needed for this ARM template. Is this reserved for future use?

File: sdk/monitor/test-resources.json (hunk 3 above)
Comment (pakrym) [inline sdk/monitor/test-resources.json:14]: We shouldn't need the default value here or in the `testApplicationOid`. The new-testresources script would always provide one.

**Comment by TimothyMothra** [comment]
This PR is on hold while making other changes to the Exporter.

**Comment by TimothyMothra** [comment]
After renaming the project, there were too many conflicts to resolve. I will open a new PR

**Pull request closed without merging**